python/meep.i

```diff
@@ -508,15 +508,26 @@ kpoint_list get_eigenmode_coefficients_and_kpoints(meep::fields *f, meep::dft_fl
 }
 
 PyObject *_get_array_slice_dimensions(meep::fields *f, const meep::volume &where, size_t dims[3],
-                                      bool collapse_empty_dimensions, bool snap_empty_dimensions) {
+                                      bool collapse_empty_dimensions, bool snap_empty_dimensions, 
+                                      meep::component cgrid = Centered, PyObject *min_max_loc = NULL) {
     meep::direction dirs[3] = {meep::X, meep::X, meep::X};
-    int rank = f->get_array_slice_dimensions(where, dims, dirs, collapse_empty_dimensions, snap_empty_dimensions);
+
+    meep::vec min_max_loc_vec[2];
+    meep::vec* min_max_loc_vec_ptr = min_max_loc_vec;
+    if (!min_max_loc) min_max_loc_vec_ptr = NULL;
+    
+    int rank = f->get_array_slice_dimensions(where, dims, dirs, collapse_empty_dimensions, snap_empty_dimensions, min_max_loc_vec_ptr, 0, cgrid);
 
     PyObject *py_dirs = PyList_New(3);
     for (Py_ssize_t i = 0; i < 3; ++i) {
         PyList_SetItem(py_dirs, i, PyInteger_FromLong(static_cast<int>(dirs[i])));
     }
 
+    if (min_max_loc){
+        PyList_Append(min_max_loc, vec2py(min_max_loc_vec[0],true));
+        PyList_Append(min_max_loc, vec2py(min_max_loc_vec[1],true));
```

Comment:
This seems like it has a memory leak — it needs to call `Py_DECREF` on the `vec2py` result after it is done with it.

(Actually, skimming over this file, it seems like a lot of the `vec2py` calls leak memory in this way.)

Reply:
(The `PyList_Append` documentation does not say that it steals the reference to the argument.  I double-checked and the [source code](https://github.com/python/cpython/blob/5241e189e77972d3a07acbbb3f0c0cbc2aeeb681/Objects/listobject.c#L313) does indeed increment the reference.  So we need to decrement the reference after appending or the vector will never be freed.)

Reply:
Should be fixed now